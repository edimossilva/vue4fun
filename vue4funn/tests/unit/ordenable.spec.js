import { shallowMount } from '@vue/test-utils'
import Ordenable from '@/components/shared/ordenable.vue'

const growerList = [
    { name: 'pedro' },
    { name: 'ana'  },
    { name: 'maria'}
  ]

const ascGrowerList = [
  { name: 'ana' },
  { name: 'maria' },
  { name: 'pedro' }
]

const dscGrowerList = [
  { name: 'pedro' },
  { name: 'maria' },
  { name: 'ana' }
]

const propertyTitle = 'Nome';

describe('Ordenable.vue', () => {

  let wrapper;

  beforeEach(() => {
    wrapper = shallowMount(Ordenable, {
      propsData: {
        list: growerList,
        propertyTitle: propertyTitle,
        propertyName: 'name'
      }
    });
  });

  it('shows up label', () => {
    const upLabel = `${propertyTitle} - ▲`;
    expect(wrapper.text()).toMatch(upLabel)
  });


  describe('after first click', () => {

    it('shows down label', () => {
      const downLabel = `${propertyTitle} - ▼`;
      wrapper.find('th').trigger('click')

      expect(wrapper.text()).toMatch(downLabel)
    });

    it('sort asc the list', () => {
      wrapper.find('th').trigger('click')

      expect(wrapper.props('list')).toEqual(ascGrowerList)
    });

    describe('after second click', () => {
      it('sort dsc the list', () => {
        wrapper.find('th').trigger('click')
        wrapper.find('th').trigger('click')

        expect(wrapper.props('list')).toEqual(dscGrowerList)
      });
    });

    it("has the expected html structure", () => {
      expect(wrapper.element).toMatchSnapshot();
    });
  });
});
