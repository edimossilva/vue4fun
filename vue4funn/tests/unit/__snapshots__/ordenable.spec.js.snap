// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Ordenable.vue after first click has the expected html structure 1`] = `
<th>
   Nome - â–² 
</th>
`;
